uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/annual-entitlements.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/absence-tab/modules/components',
+  'common/models/contact'
+], function (_, moment, components) {
+  components.component('annualEntitlements', {
+    bindings: {
+      contactId: '<'
+    },
+    templateUrl: ['settings', function (settings) {
+      return settings.pathTpl + 'components/annual-entitlements.html';
+    }],
+    controllerAs: 'entitlements',
+    controller: ['$log', '$q', 'AbsenceType', 'AbsencePeriod', 'Entitlement', 'Contact', controller]
+  });
+
+  function controller ($log, $q, AbsenceType, AbsencePeriod, Entitlement, Contact) {
+    $log.debug('Component: annual-entitlements');
+
+    var vm = {};
+
+    vm.absenceTypes = [];
+    vm.absencePeriods = [];
+    vm.contactId = this.contactId;
+    vm.contacts = [];
+    vm.entitlements = [];
+    vm.loaded = { absencePeriods: false };
+
+    (function init () {
+      return $q.all([
+        loadAbsenceTypes(),
+        loadEntitlements()
+      ])
+      .then(function () {
+        return loadCommentsAuthors();
+      })
+      .then(function () {
+        return loadAbsencePeriods();
+      });
+    })();
+
+    /**
+     * Shows a comment to the entitlement
+     *
+     * @param {object} comment
+     */
+    vm.showComment = function (comment) {
+      /*
+       * @TODO There is no support for footer in CRM.alert at the moment.
+       * This code should be refactored as soon as CRM.alert supports footer.
+       * At the moment the footer is constructed via rich HTML directly via body text
+       */
+      var text = comment.message +
+        '<br/><br/><strong>Last updated:' +
+        '<br/>By: ' + comment.author_name +
+        '<br/>Date: ' + moment(comment.date).format('DD/M/YYYY HH.mm') +
+        '</strong>';
+
+      CRM.alert(text, 'Calculation comment:', 'error');
+    };
+
+    /**
+     * Loads absence periods
+     *
+     * @return {Promise}
+     */
+    function loadAbsencePeriods () {
+      return AbsencePeriod.all()
+        .then(function (absencePeriods) {
+          setAbsencePeriodsProps(absencePeriods);
+        });
+    }
+
+    /**
+     * Loads absence types
+     *
+     * @return {Promise}
+     */
+    function loadAbsenceTypes () {
+      return AbsenceType.all()
+        .then(function (absenceTypes) {
+          vm.absenceTypes = absenceTypes;
+        });
+    }
+
+    /**
+     * Loads entitlements comments authors
+     *
+     * @return {Promise}
+     */
+    function loadCommentsAuthors () {
+      var authorsIDs = _.uniq(_.map(vm.entitlements, function (entitlement) {
+        return entitlement.comment_author_id;
+      }));
+
+      return Contact.all({ id: { 'IN': authorsIDs } })
+        .then(function (contacts) {
+          vm.contacts = contacts.list;
```

Comment:
Index contacts by `contact_id` so that you dont need to loop when you want to fetch the author_name, instead 
`contacts[leave.comment_author_id]` can be used.

`contacts = _.indexBy(contacts.list, 'contact_id')`